SmartDeviceLink/SDLKeyboardDelegate.h

```diff
@@ -0,0 +1,90 @@
+//
+//  SDLKeyboardDelegate.h
+//  SmartDeviceLink
+//
+//  Created by Joel Fischer on 5/21/18.
+//  Copyright © 2018 smartdevicelink. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+
+#import "SDLKeyboardEvent.h"
+
+@class SDLKeyboardProperties;
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ This handler is called when you wish to update your autocomplete text in response to the user's input
+
+ @param updatedAutocompleteText The new autocomplete text to use
+ */
+typedef void(^SDLKeyboardAutocompleteCompletionHandler)(NSString  *_Nullable updatedAutocompleteText);
+
+/**
+ This handler is called when you wish to update your keyboard's limitedCharacterSet in response to the user's input
+
+ @param updatedCharacterSet The new set of characters to use
+ */
+typedef void(^SDLKeyboardCharacterSetCompletionHandler)(NSArray<NSString *> *_Nullable updatedCharacterSet);
+
+@protocol SDLKeyboardDelegate <NSObject>
+
+//
+
+/**
+ The keyboard session completed with some input.
+
+ This will be sent upon ENTRY_SUBMITTED or ENTRY_VOICE. If the event is ENTRY_VOICE, the user requested to start a voice session in order to submit input to this keyboard. This MUST be handled by you. Start an Audio Pass Thru session if supported.
+
+ @param inputText The submitted input text on the keyboard
+ @param source ENTRY_SUBMITTED if the user pressed the submit button on the keyboard, ENTRY_VOICE if the user requested that a voice session begin
+ */
+- (void)userDidSubmitInput:(NSString *)inputText withEvent:(SDLKeyboardEvent)source;
+
+/**
+ The keyboard session aborted.
+
+ This will be sent if the keyboard event ENTRY_CANCELLED or ENTRY_ABORTED is sent
+
+ @param event ENTRY_CANCELLED if the user cancelled the keyboard input, or ENTRY_ABORTED if the system aborted the input due to a higher priority event
+ */
+- (void)keyboardDidAbortWithReason:(SDLKeyboardEvent)event;
+
+@optional
+/**
+ Implement this in order to provide a custom keyboard configuration to just this keyboard. To apply default settings to all keyboards, see SDLScreenManager.keyboardConfiguration
+
+ @return The custom keyboard configuration to use.
+ */
+- (SDLKeyboardProperties *)customKeyboardConfiguration;
```

Comment:
`customKeyboardConfiguration` is marked as _optional_ but if I do not implement, the app crashes when I use `presentKeyboardWithInitialText:delegate:`.

It looks like line 64 of the `SDLPresentKeyboardOperation` class does not check if `customKeyboardConfiguration` is implemented.

Reply:
Should be fixed